File: docs/basic/code/bubble-sort/bubble-sort_1.py
# --8<-- [start:core]
def bubble_sort(a, n):
    flag = True
    while flag:
        flag = False
        for i in range(1, n):
            if a[i] > a[i + 1]:
                flag = True
                a[i], a[i + 1] = a[i + 1], a[i]


# --8<-- [end:core]
if __name__ == "__main__":
    n = int(input())
    a = [0] + [int(x) for x in input().split()]

    bubble_sort(a, n)
    print(" ".join(map(str, a[1:])))

File: basic/code/bubble-sort/bubble-sort_1.cpp
// 假设数组的大小是 n + 1，冒泡排序从数组下标 1 开始
void bubble_sort(int *a, int n) {
  bool flag = true;
  while (flag) {
    flag = false;
    for (int i = 1; i < n; ++i) {
      if (a[i] > a[i + 1]) {
        flag = true;
        int t = a[i];
        a[i] = a[i + 1];
        a[i + 1] = t;
      }
    }
  }
}
